### PR TITLE
[ENG-2391] Add moderators acceptance tests.

### PR DESCRIPTION
**Pull request by adlius**
- Ticket: [ENG-2391]
- Feature flag: n/a

## QA Notes

[ENG-2391]: https://openscience.atlassian.net/browse/ENG-2391

### CODE REVIEW

File: tests/engines/registries/acceptance/branded/moderation/moderators-test.ts [hunk 1]
```diff
@@ -63,4 +68,129 @@ module('Registries | Acceptance | branded.moderation | moderators', hooks => {
...
+        await selectChoose('[data-test-select-permission]', 'Moderator');
+        await click('[data-test-confirm-add-moderator-button]');
+        assert.dom('#toast-container', document as any).hasTextContaining(
+            t(
+                'registries.moderation.moderators.addedNewModeratorError',
+                { permission: 'moderator' },
+            ),
+            'Toast error message for adding user as moderator failure',
+        );
+        await timeout(5000);
```
Comment (fabmiz) [inline tests/engines/registries/acceptance/branded/moderation/moderators-test.ts:173]: what's this timeout for? would `await settled()` do?

Comment (futa-ikeda) [inline tests/engines/registries/acceptance/branded/moderation/moderators-test.ts:173]: Do we need these timeouts? Are error toasts and success toasts different in some way?

Comment (adlius) [reply tests/engines/registries/acceptance/branded/moderation/moderators-test.ts:173]: This is waiting for the previous error toast to go away. Somehow the app doesn't show more than one error toasts at the same time.

File: tests/engines/registries/acceptance/branded/moderation/moderators-test.ts [hunk 2]
```diff
@@ -63,4 +68,129 @@ module('Registries | Acceptance | branded.moderation | moderators', hooks => {
...
+        }), 400);
+        server.put('/providers/registrations/:parentID/moderators/:id/', () => ({
+            errors: [{ detail: '' }],
+        }), 400);
+        server.patch('/providers/registrations/:parentID/moderators/:id/', () => ({
+            errors: [{ detail: '' }],
+        }), 400);
+        server.del('/providers/registrations/:parentID/moderators/:id/', () => ({
+            errors: [{ detail: '' }],
+        }), 400);
```
Comment (futa-ikeda) [inline tests/engines/registries/acceptance/branded/moderation/moderators-test.ts:153]: Just wanted to comment on this for future reference. Good to overwrite mirage route handlers and request failures as well 👀

**Review by futa-ikeda — APPROVED** [review]

**Review by fabmiz — APPROVED** [review]

**Pull request merged**